File: src/minim/controller/table/chronicles/werewolf/Werewolf.java
package minim.controller.table.chronicles.werewolf;

import java.util.ArrayList;
import java.util.List;

import minim.controller.table.SimpleTable;
import minim.controller.table.Table;

public class Werewolf extends Table {
	public static final Table AUSPICE = new SimpleTable("Werewolf (auspice)",
			List.of("Warrior", "Visinary", "Walker-between", "Spirit master", "Stalker"));

	public Werewolf() {
		super("Werewolf");
	}

	@Override
	public void build() {
		// don't
	}

	@Override
	public String roll() {
		var lines = new ArrayList<String>();
		lines.add("Tribe: " + Tribe.INSTANCE.roll().toLowerCase());
		lines.add("Auspice: " + AUSPICE.roll().toLowerCase());
		return String.join(".\n", lines);
	}
}

File: web/script/control/table/darkness/werewolf.js
import * as table from '../table.js'

const AUSPICE = new table.Table("Darkness, werewolf, auspice",
    ["Warrior", "Visinary", "Walker-between", "Spirit master", "Stalker"])

const FORSAKEN = new table.Table("Darkness, werewolf, tribe, forsaken",
    ["Blood talons", "Bone shadows", "Hunters in darkness", "Iron masters", "Storm lords"])
const PURE = new table.Table("Darkness, werewolf, tribe, pure",
    ["Fire-touched", "Ivory claws", "Predator kings"])
const OTHER = new table.Table("Darkness, werewolf, tribe, other",
    ["Ghost wolves", "Bale hounds"])

const TRIBES = new Map();
for (let t of [FORSAKEN, PURE, OTHER]) TRIBES.set(t.name, t);

class Tribe extends table.Table {
	constructor() {
		super("Darkness, werewolf, tribe");
		this.add(FORSAKEN, 2);
		this.add(PURE, 3);
		this.add(OTHER, 1);
	}
}

var tribe=new Tribe()

class Werewolf extends table.Table {
	constructor() {
		super("Darkness, werewolf");
	}

	roll() {
		let lines = [
      "Tribe: " + tribe.roll().toLowerCase(),
      "Auspice: " + AUSPICE.roll().toLowerCase()
		]
		return lines.join('<br/>')
	}
}

export var tables=[AUSPICE,FORSAKEN,PURE,OTHER,tribe,new Werewolf()]
